pkg/controllers/vdb/serviceaccount_reconciler.go

```diff
@@ -0,0 +1,205 @@
+/*
+ (c) Copyright [2021-2023] Open Text.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ You may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package vdb
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	vapi "github.com/vertica/vertica-kubernetes/api/v1beta1"
+	"github.com/vertica/vertica-kubernetes/pkg/builder"
+	"github.com/vertica/vertica-kubernetes/pkg/controllers"
+	"github.com/vertica/vertica-kubernetes/pkg/iter"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+// ServiceAccountReconciler will handle generation of the service account
+type ServiceAccountReconciler struct {
+	VRec *VerticaDBReconciler
+	Vdb  *vapi.VerticaDB // Vdb is the CRD we are acting on.
+	Log  logr.Logger
+}
+
+func MakeServiceAccountReconciler(vdbrecon *VerticaDBReconciler, log logr.Logger, vdb *vapi.VerticaDB) controllers.ReconcileActor {
+	return &ServiceAccountReconciler{
+		VRec: vdbrecon,
+		Vdb:  vdb,
+		Log:  log.WithName("ServiceAccountReconciler"),
+	}
+}
+
+// Reconcile will ensure that a serviceAccount, role and rolebindings exists for
+// the vertica pods.
+func (s *ServiceAccountReconciler) Reconcile(ctx context.Context, _ *ctrl.Request) (ctrl.Result, error) {
+	rbacFinder := iter.MakeRBACFinder(s.VRec.Client, s.Vdb)
```

Comment:
Can users still specify their own sa?

Reply:
For now they can't. But that was the follow-on story I mentioned in scrum.